src/test/java/edu/wpi/grip/core/StepTest.java

```diff
@@ -34,6 +34,18 @@ public void testStep() {
     }
 
     @Test
+    public void testSocketDirection() {
+        Step step = new Step(eventBus, addition);
+        Socket<Double> a = (Socket<Double>) step.getInputSockets()[0];
+        Socket<Double> b = (Socket<Double>) step.getInputSockets()[1];
+        Socket<Double> c = (Socket<Double>) step.getOutputSockets()[0];
+
+        assertEquals(Socket.Direction.INPUT, a.getDirection());
+        assertEquals(Socket.Direction.INPUT, b.getDirection());
+        assertEquals(Socket.Direction.OUTPUT, c.getDirection());
```

Comment:
You haven't proved anything about the underlying implementation with this test.
This basically proves that when you set a variable you can get it back out again the same way.